### ISSUE TITLE
Thoughts on False Reports

### ISSUE THREAD

**Issue by HendX** [CrunchyBagel/TracePrivately#11]
There's no mechanism in the current prototype to prevent false reports.

This thread is for discussion of strategies to address this.

**Comment by HendX** [comment]
1. Every submission needs to be approved on the server side
2. This would mean the assessor would require additional data with which to approve/deny
3. Perhaps a photo of a positive test
4. Presumably a test has some kind of unique identifier that can be subsequently verified

**Comment by tatey** [comment]
You could also use that unique identifier to guard another set of keys being associated with a positive test. E.g. I share the photo with someone and they submit the same photo.

**Comment by HendX** [comment]
Agreed. For now I've just added extra options to the server. New submissions are created in a pending state and require manual approval (using supplied command line tools) before they will be returned by the `/api/infected` call.

**Comment by alexcfaber** [comment]
It depends on the use case (and probably on the country implementing it). 

If a local health organisation would be in control of the backend they could also be in control of approving. 

One way that could work is that when a user tries to report themselves as infected it generates a unique token which they can report to their health organisation. That organisation can generate a token based on that token and the outcome of that gets communicated back to the user. When they report themselves as infected they can confirm it using that code. 

That might be a bit too complicated for the user though. 

Another way as you mention could be a pending state, you would still need to connect an actual infected person with that report though; so at that point you could have a generated token to confirm it. But the user doesn't have to do anything.

**Comment by HendX** [comment]
I think my plan is to add configuration options to the app that allows the submission screen to have extra fields, such as:

1) Photo(s) submission
2) Pathology test ID
3) Name / contact details (to match up the name again the test ID)

The intention would be that different organisations making use of the code could easily reconfigure the form with minimal coding.

**Comment by tatey** [comment]
FYI I have begun working on an Admin UI on the server . The idea is that a health professional can look at recent submissions and confirm a positive or negative test result. I'm hoping to expand on this UI as we work out what data is going to be submitted.

![admin_index](https://user-images.githubusercontent.com/19860/80102069-a0b4ce00-85b5-11ea-9738-791e1131a40b.png) 

![admin_show](https://user-images.githubusercontent.com/19860/80102058-9c88b080-85b5-11ea-9b6c-00fcae247bbc.png)

**Comment by HendX** [comment]
Ah very nice. I still haven't implemented handling of negative keys in the iOS app. Getting to it.
